action.yml

```diff
@@ -0,0 +1,44 @@
+name: Sync Salmon Directory
+description: Syncs a directory to a Google Cloud bucket using rclone.
+author: 'deephaven'
+inputs:
+  source:
+    required: true
+    type: string
+    description: 'The source directory to sync.'
+  destination:
+    required: true
+    type: string
+    description: 'The destination directory to sync. Relative to the bucket.'
+  credentials:
+    required: true
+    type: string
+    description: 'The Google Cloud credentials.'
+
+runs:
+  using: "composite"
+  # this action only runs if the input credentials match the secret
+  if: ${{ secrets.DOCS_GOOGLE_CLOUD_CREDENTIALS == inputs.credentials }}
+  steps:
+    - name: Setup rclone
+      shell: bash
+      run: |
+        sudo apt-get update
+        sudo apt-get install -y rclone=1.53.3-4ubuntu1.22.04.2
+
+        mkdir -p $HOME/.config
+        mkdir -p $HOME/.config/rclone
+
+        cat << EOF > $HOME/.config/rclone/rclone.conf
+        [docs]
+        type = google cloud storage
+        service_account_file = $HOME/credentials.json
+        project_number = ${{ secrets.DOCS_GOOGLE_CLOUD_PROJECT_NUMBER }}
+        bucket_policy_only = true
+        EOF
+        
+        echo ${{ inputs.credentials }} | base64 -d > $HOME/credentials.json
```

Comment:
If possible, I think this should be an env variable that we set in the action. See my other comment about using secrets in conditions which also mentions some stuff about using secrets w/ CLI

Reply:
given this is redacted as shown in https://github.com/deephaven/salmon-sync/pull/1#discussion_r1805257855 is this fine as is or should I still modify it?

Reply:
GitHub recommends using environment variables whenever possible [here](https://docs.github.com/en/actions/security-for-github-actions/security-guides/using-secrets-in-github-actions#using-secrets-in-a-workflow). I think there's a potential concern if the action runners were compromised or had bad actors they could potentially steal info out of the system monitor since the actual process will print w/ the values on the host machine.

I think you can simplify this a bit by instead using `service_account_credentials` which you can conveniently set w/ the `RCLONE_GCS_SERVICE_ACCOUNT_CREDENTIALS` environment variable (docs [here](https://rclone.org/googlecloudstorage/#gcs-service-account-credentials)). So set that env variable in the action to `${{ inputs.credentials }}` and I think you're good (make sure it still obscures the secret in action logs, but I don't know why it wouldn't).

There are also env variables for the other secrets on that docs page like the project number we could set.

You could also use `rclone config` CLI instead of making the file. [Docs](https://rclone.org/commands/rclone_config_create/). I think the command would be `rclone config create docs "google cloud storage" bucket_policy_only true`. As long as the 2 env variables are set for project number and service account credentials, then it seems to make the config locally for me and it should read the env variables at runtime instead of writing to the file

Reply:
I just went ahead and moved everything to environment variables and it seems to work. The only exception is the credential because it has to b base64 decoded and I don't think there's a better way to do that. Considering github does this in an [example](https://docs.github.com/en/enterprise-server@3.10/actions/security-for-github-actions/security-guides/using-secrets-in-github-actions#storing-base64-binary-blobs-as-secrets) I think it should be fine